drivers/dma/dma_sedi.c

```diff
@@ -0,0 +1,582 @@
+	}
+	/* configurate dma width of source data*/
+	ret = width_index(config->source_data_size, &temp);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_SR_TRANS_WIDTH, temp);
+
+	/* configurate dma width of destination data*/
+	ret = width_index(config->dest_data_size, &temp);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_DT_TRANS_WIDTH, temp);
+
+	/* configurate dma burst size*/
+	ret = burst_index(config->source_burst_length, &temp);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_BURST_LENGTH, temp);
+	return 0;
+
+INVALID_ARGS:
+	return ret;
+}
+
+static int dma_sedi_chan_config(const struct device *dev, uint32_t channel,
+				struct dma_config *config)
+{
+	if ((dev == NULL) || (channel >= DEV_CFG(dev)->chn_num)
+		|| (config == NULL)
+		|| (config->block_count != 1)) {
+		goto INVALID_ARGS;
+	}
+
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *local_config = &(data->dma_configs[channel]);
+
+	memcpy(local_config, config, sizeof(struct dma_config));
+
+	local_config->head_block =
+		alloc_and_cpy_blk_nodes(config->head_block);
+
+	if (local_config->head_block == NULL) {
+		return -1;
+	}
+
+	/* initialize the dma controller, following the sedi api*/
+	sedi_dma_event_cb_t cb = dma_handler;
+
+	sedi_dma_init(info->instance, (int)channel, cb, NULL);
+
+	return 0;
+
+INVALID_ARGS:
+	return -1;
+}
+
+static int dma_sedi_reload(const struct device *dev, uint32_t channel,
+			      uint64_t src, uint64_t dst, size_t size)
+{
+	if ((dev == NULL) || (channel >= DEV_CFG(dev)->chn_num)) {
+		LOG_ERR("dma reload failed for invalid args");
+		return -ENOTSUP;
+	}
+
+	int ret = 0;
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *config = &(data->dma_configs[channel]);
+	struct dma_block_config *block_config;
+
+	if ((config == NULL) || (config->head_block == NULL)) {
+		LOG_ERR("dma reload failed, no config found");
+		return -ENOTSUP;
+	}
+	block_config = config->head_block;
+
+	if ((config->block_count == 1) || (block_config->next_block == NULL)) {
+		block_config->source_address = src;
+		block_config->dest_address = dst;
+		block_config->block_size = size;
+	} else {
+		LOG_ERR("no reload support for multi-linkedlist mode");
+		return -ENOTSUP;
+	}
+	return ret;
+}
+
+static int dma_sedi_start(const struct device *dev, uint32_t channel)
+{
+	if ((dev == NULL) || (channel >= DEV_CFG(dev)->chn_num)) {
+		LOG_ERR("dma transferring failed for invalid args");
+		return -ENOTSUP;
+	}
+
+	int ret = -1;
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *config = &(data->dma_configs[channel]);
+	struct dma_block_config *block_config = config->head_block;
+	uint64_t src_addr, dst_addr;
+
+	if (k_is_in_isr()) {
```

Comment:
remove is_in_isr() checking.
previously it's for paging which has already switched to use SEDI dma APIs already.

Reply:
Done